content/articles/docker-push-for-publishing-images-to-docker-hub/index.md

```diff
@@ -0,0 +1,125 @@
+Now that we have created our Dockerfile before we push it to an online repository, let's create a repository within Docker.
+If you're well-acquainted with the way Github works then Docker hub isn't that different from it. So head over to [Docker Hub](https://hub.docker.com/) and register an account. After signup, click the `Repositories` tab in the navbar, you'll see a form like the one below:
+
+![repository](engineering-education/docker-push-for-publishing-images-to-docker-hub/repo.jpg)
+
+Create a repository called `docker-push` as that is the example we'll be using throughout the article. Now that our repository is set,  let's proceed and create an image from Docker and push it to the repository we created earlier.
+
+### Build Docker Image using Docker Hub and for organization
+To be able to build an image in Docker, the command below is used:
+ ```
+ docker build -t username/repository_name .
+ ```
+
+The `-t` flag helps when dealing with various images in identifying which name they belong to. The `username` is your Docker hub name and the `repository_name` in this case is `docker-push` the repository we created earlier. We add a period `.` when in the folder of the Dockerfile. With this hindsight, let's proceed and build our Docker image by executing the command changing the username with yours as it appears in the Docker hub and the repository with `docker-push` as displayed below:
+
+```
+docker build -t bullet08/docker-push .
+```
+
+For more about Docker Hub commands, check the documentation [here](https://docs.docker.com/docker-hub/repos/).
+
+The same approach is used when building Docker images for organizations, you change the username with the organization's account name and Docker hub repository of the organization.
+
+### Pushing Docker image
+Before we push the Docker image we need to log into the Docker hub. We can do this effortlessly on the command-line using the command shown below:
+
+```
+oruko@oruko-ThinkPad-T520:~/Documents/docker$ docker login
+username: bullet08
+password:
+Login Succeeded
+```
+
+Once validated, we can then push our container to the Docker hub.
+We do this using the commands below:
+
+```
+docker push bullet08/docker-push
+```
+With that done, our Docker image is now available in Docker Hub. You can view it by visiting your repository.
+
+### Pushing to a Non-Docker-Hub registry
+A Docker registry is a memory board and a disseminated system for Docker images that enable Docker users to pull built images to their local and push new images to the registry when for authorized users. This far, we have conferred how to build a Docker image using our username. As this is important for individuals using the Docker Hub registry service, the procedures change a bit when pushing to the Docker-Hub registry. We append our registry to our username when building for the non-Docker-Hub-Registry. The example below shows an example of building an image for a registry at `registry.test.com`.
```

Comment:
Try making your sentences shorter. That definition of a Docker registry is too long.